src/Build/Instance/TaskRegistry.cs

```diff
@@ -1189,12 +1247,36 @@ internal RegisteredTaskRecord(string registeredName, AssemblyLoadInfo assemblyLo
                 {
                     _parameterGroupAndTaskBody = new ParameterGroupAndTaskElementRecord();
                 }
+
+                _definingFileFullPath = containingFileFullPath;
             }
 
             private RegisteredTaskRecord()
             {
             }
 
+            public bool GetIsCustom()
```

Comment:
this is very confusing

Reply:
Added claryfying comment - please hav a look whether it feels sufficient.

Reply:
the comment clarifies the confusion, but the name is still a little confusing. 
`IsCustom` probably not a great name because it is not a set bool, you still need to calculate it.

nit: rename to smth like or ComputeIfCustom, CheckIfCustom or DetermineIfCustom

Reply:
Makes sense! - renamed